### PR TITLE
CI: update Fedora to 34

### PR DESCRIPTION
**Pull request by AkihiroSuda**
(Marked as a draft until https://github.com/kubernetes-sigs/kind/pull/2230 `make podman resolve short names ` gets merged)

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Podman failing

```
using podman due to KIND_EXPERIMENTAL_PROVIDER
enabling experimental podman provider
Creating cluster "kind" ...
 • Ensuring node image (kindest/node:v1.20.2) 🖼  ...
DEBUG: podman/images.go:67] Pulling image: kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7 ...
DEBUG: podman/images.go:73] Trying again to pull image: "kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" ... command "podman pull kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" failed with error: exit status 125
DEBUG: podman/images.go:73] Trying again to pull image: "kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" ... command "podman pull kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" failed with error: exit status 125
DEBUG: podman/images.go:73] Trying again to pull image: "kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" ... command "podman pull kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" failed with error: exit status 125
DEBUG: podman/images.go:73] Trying again to pull image: "kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" ... command "podman pull kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" failed with error: exit status 125
 ✗ Ensuring node image (kindest/node:v1.20.2) 🖼
ERROR: failed to create cluster: failed to pull image "kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7": command "podman pull kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7" failed with error: exit status 125

Command Output: Error: error getting default registries to try: short-name resolution enforced but cannot prompt without a TTY
```
https://github.com/kubernetes-sigs/kind/pull/2220/checks?check_run_id=2455179617

**Comment by tao12345666333** [comment]
/ok-to-test 

Thanks!

**Comment by AkihiroSuda** [comment]
Not sure whether Podman failure is a real issue, or just hitting Docker Hub rate limit

**Comment by AkihiroSuda** [comment]
Seems to require some config to shut up prompt
```console
$ podman pull kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7
? Please select an image: 
  ▸ registry.fedoraproject.org/kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7
    registry.access.redhat.com/kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7
    docker.io/kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7
    quay.io/kindest/node@sha256:15d3b5c4f521a84896ed1ead1b14e4774d02202d5c65ab68f30eeaf310a3b1a7
```

**Comment by aojea** [comment]
related to https://github.com/kubernetes-sigs/kind/issues/2186#issue-852466449
/lgtm

**Comment by AkihiroSuda** [comment]
/test pull-kind-e2e-kubernetes-1-19
--

**Comment by AkihiroSuda** [comment]
@BenTheElder Could you take a look?

File: hack/ci/Vagrantfile [hunk 1]
```diff
@@ -19,10 +19,14 @@ Vagrant.configure("2") do |config|
     set -eux -o pipefail
     dnf install -y golang-go make kubernetes-client podman
 
-    # The moby-engine package (v19.03) included in Fedora 33 does not support cgroup v2.
-    # So we need to install Docker 20.10 (or later) from the upstream.
     curl -fsSL https://get.docker.com | sh
     systemctl enable --now docker
+
+    # Configure Podman to resolve `kindest/node` to `docker.io/kindest/node`.
```
Comment (BenTheElder) [inline hack/ci/Vagrantfile:25]: We need to patch this in kind. There's an open issue discussing.

**Review by BenTheElder — COMMENTED** [review]
@aojea has approval permission as well. I haven't been in every PR since we added more maintainers.

I'm not sure we should configure podman in a nonstandard way that users will run into versus patching the podman driver to deal with it.

**Comment by aojea** [comment]
> I'm not sure we should configure podman in a nonstandard way that users will run into versus patching the podman driver to deal with it.

That is why I was waiting for you , I understood from your comment in the issue that you expected users to override https://github.com/kubernetes-sigs/kind/issues/2186#issuecomment-816322290 , 
but we know that will come to use like with all the fedora firewalld/nftables/... issues 

Should we patch then in kind?

**Comment by BenTheElder** [comment]
> That is why I was waiting for you , I understood from your comment in the issue that you expected users to override #2186 (comment) ,

_docker_ users can set a mirror for this.

> Should we patch then in kind?

see: https://github.com/kubernetes-sigs/kind/issues/2186#issuecomment-816316163

> We can do our own short name resolution in the podman compatibly layer.

The podman node driver should deal with this as kubernetes does and resolve short names to be as expected.

**Comment by tao12345666333** [comment]
> The podman node driver should deal with this as kubernetes does and resolve short names to be as expected.

@BenTheElder @aojea  let me give it a try, I will submit a PR ASAP.

**Comment by AkihiroSuda** [comment]
Carried as https://github.com/kubernetes-sigs/kind/pull/2285

**Pull request closed without merging**